src/Promact.TestCaseManagement.Core/Controllers/ModuleController.cs

```diff
@@ -0,0 +1,157 @@
+                return NotFound();
+            }
+
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(module);
+        }
+
+        /// <summary>
+        /// Method to add module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPost("{projectId}/module")]
+        public async Task<IActionResult> CreateModuleAsync(int projectId, [FromBody]ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+            var module = Mapper.Map<Module>(moduleAC);
+            module.ProjectId = projectId;
+
+            return Ok(await _iModuleRepository.AddModuleAsync(module));
+        }
+
+        /// <summary>
+        /// Method to update module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPut("{projectId}/module/{id}")]
+        public async Task<IActionResult> UpdateModuleAsync(int projectId, int id, [FromBody] ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            Mapper.Map(moduleAC, module);
+            await _iModuleRepository.UpdateModuleAsync(module);
+
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Method to delete module from the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <returns></returns>
+        [HttpDelete("{projectId}/module/{id}")]
+        public async Task<IActionResult> DeleteModule(int projectId, int id)
+        {
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            await _iModuleRepository.DeleteModuleAsync(module);
```

Comment:
Apologies, due to my lack of knowledge this discussion is ongoing. 
Two things are here I would like to address.
1) Your point is valid that Authorization is missing. Agreed, But we have thought it to integrate it once the client side is ready. But will do it now.
2) regarding project exist or not. Your point is totally valid that project id is a foreign key. So it will ensure it. But what I like to address here is that if I pass project id which does not exist in a database, then it will give me 500 internal server error if I do not check project exist or not. Which is according to me gives a wrong status code to an end user. In this case, the status code should have 404 as per my understanding. I would like to know what you are thinking about this scenario. Is 500 a good choice or 404?